core/src/main/java/org/jdbi/v3/core/statement/ParsedParameters.java

```diff
@@ -78,4 +78,22 @@ public String toString() {
             + ", parameterNames=" + parameterNames
             + '}';
     }
+
+    /**
+     * A static factory for {@link ParsedParameters} objects. It takes two
+     * arguments. The first argument is a boolean flag which indicates if
+     * statement, to which this object refers to, is using positional
+     * (<code>true</code> value) or named parameters (a <code>false</code>
+     * value). The second argument is a list which contains parameters
+     * available in SQL. In case when positional parameters are used the
+     * names list should contain only positional parameters, or more
+     * strictly, a "?" characters.
+     *
+     * @param positional true if statement uses positional parameters, false otherwise
+     * @param names the parameter names from SQL statement
+     * @return New {@link ParsedParameters} object
+     */
+    public static ParsedParameters of(boolean positional, List<String> names) {
```

Comment:
I think API-wise (without considering functionality yet) this factory could be better: either the params are positional and you need a List (maybe a List/Set hybrid for uniqueness?) of names, or they are positional and you need to know how many (as an int) there are. I'd split this into two, more expressive methods.

Reply:
Having `Set` instead of a list is a no-go IMO because it can't be accessed by index. In regards to your other comments I may suggest the following:

```java
 public static ParsedParameters positional(int count)
 public static ParsedParameters named(List<String> names)
```

Where `positional(int count)` will create list with a given size and fill it with "?" characters. Please let me know if this is ok.

Reply:
Yeah those are the factory methods I meant.

Set wouldn't work, but e.g. NavigableSet or LinkedHashSet would. I have a feeling the other guys will say to just use a List instead of those exotic types though, so it's fine.

Reply:
Come to think of it, uniqueness probably isn't even wanted there because the same name should be able to appear there multiple times, shouldn't it?

Reply:
Yes. Same name can appear multiple times on a list and parameter index is mapped to its position in SQL. That's why `Set` cannot be used in the first place.

Reply:
Done.